### PR TITLE
WIP: initial commit of route_func and common library

### PR DESCRIPTION
**Pull request by rasibley**
This PR includes the networking common library and route_func test case. Currently we use RhtsRequires to pull in the library in the Makefile, but instead we need to port the library and then fetch it in the metadata file using 'Requires=networking/common'. I'm still waiting on review from the test maintainers, however would like initial feedback while we wait. @veruu @idorax @ftjahjad

### CODE REVIEW

**Comment by ftjahjad** [comment]
@rasibley 
I think is very clean to have  "networking common" library. :100: 

seeing another "networking tunnel common library" for tunnel  basic test that I need to port .

File: networking/common/lib/protocol/netns.sh [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	CONTROLLER_IFACE="br_lan"
+	CONTROLLER_ADDR4="192.168.1.254"
+	CONTROLLER_ADDR6="2001::254"
+	run "ip addr add $CONTROLLER_ADDR4/24 dev $CONTROLLER_IFACE"
+	run "ip addr add $CONTROLLER_ADDR6/64 dev $CONTROLLER_IFACE"
+
+	# Setup Server
+	vcommon server "rm -f /tmp/test_*; $VTOPO; get_test_iface"
+	SERVER_IFACE=$(tail -n1 /tmp/test_iface | tr -d '\r\n')
+	[ ! "$SERVER_IFACE" ] && test_fail "No SERVER IFACE" && return 1
```
Comment (veruu) [inline networking/common/lib/protocol/netns.sh]: Is this really a failure, or is it just blocking the test from running?

File: networking/common/lib/protocol/netns.sh [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	SERVER_ADDR4="192.168.1.1"
+	SERVER_ADDR6="2001::1"
+	vrun server ip link set $SERVER_IFACE up
+	vrun server ip addr add $SERVER_ADDR4/24 dev $SERVER_IFACE
+	vrun server ip addr add $SERVER_ADDR6/64 dev $SERVER_IFACE
+
+	# Setup Client
+	vcommon client "rm -f /tmp/test_*; $VTOPO; get_test_iface"
+	CLIENT_IFACE=$(tail -n1 /tmp/test_iface | tr -d '\r\n')
+	[ ! "$CLIENT_IFACE" ] && test_fail "No CLIENT IFACE" && return 1
```
Comment (veruu) [inline networking/common/lib/protocol/netns.sh]: Same here

File: networking/common/obsolete.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		else
+			echo -e "wrong option , should be 'run'\n"
+			return 1
+		fi
+	fi
+	if [ $3 ]; then
+		if [ "$3" == "check" ];then
+			if [ $LASTSTATE -ne 0 ];then
+				echo -e "Command # $1 failed\n" | tee -a "$OUTPUTFILE"
+				test_warn "Command_failed"
```
Comment (veruu) [inline networking/common/obsolete.sh]: Should this be a failure instead? Or can it mean the preparation for the test could have failed too (in which case it should abort)?

File: networking/common/include.sh [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+	fi
+	until [ $timeout -le 0 ] || [ $resend -eq 0 ]; do
+		let resend=0
+		for host in $peer_host; do
+			ssh $host "echo $HOSTNAME $message >> /tmp/sync_message" 2>/dev/null || let resend++;
+		done
+		sleep 5
+		let timeout=timeout-5
+	done
+	if [ $timeout -le 0 ]; then test_warn "rhts-sync-set $HOSTNAME $message failed"; fi
```
Comment (veruu) [inline networking/common/include.sh]: Should this also abort?

File: networking/route/route_func/common/include.sh [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+	fi
+	until [ $timeout -le 0 ] || [ $resend -eq 0 ]; do
+		let resend=0
+		for host in $peer_host; do
+			ssh $host "echo $HOSTNAME $message >> /tmp/sync_message" 2>/dev/null || let resend++;
+		done
+		sleep 5
+		let timeout=timeout-5
+	done
+	if [ $timeout -le 0 ]; then test_warn "rhts-sync-set $HOSTNAME $message failed"; fi
```
Comment (veruu) [inline networking/route/route_func/common/include.sh]: Same here

**Comment by rasibley** [comment]
Thanks @veruu  ! I have modified test_warn() function in the include.sh file to abort so it should cover all cases where where this is used (infra failures), also removed some files which shouldn't have been there.

**Comment by veruu** [comment]
As the points were resolved, let's merge this and we can do any changes later if needed.

**Pull request merged**